src/main/java/uk/gov/dwp/dataworks/provider/KMSDataKeyDecryptionProvider.java

```diff
@@ -36,9 +43,14 @@ public DecryptDataKeyResponse decryptDataKey(String dataKeyEncryptionKeyId, Stri
                     encoder.encodeToString(result.getPlaintext().array())
             );
         }
-        catch(NotFoundException | DisabledException | InvalidCiphertextException | KeyUnavailableException |
+        catch(InvalidCiphertextException ex) {
+            logger.error("Exception caught while communicating with KMS", ex);
+            throw new GarbledDataKeyError();
```

Comment:
Should we be passing out ex.getMessage() or similar here as this isn't a public system?

Reply:
Don't want to report content of error out over the wire